File: Lista/quest06/composicao.go
package main

import "fmt"

type Motor struct {
	Marca    string
	Potencia int
}

type Carro struct {
	Modelo string
	Motor  Motor
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Modelo: %s, Motor: %s, Potência: %d cv\n", c.Modelo, c.Motor.Marca, c.Motor.Potencia)
}

func main() {
	motor := Motor{Marca: "Honda", Potencia: 150}
	carro := Carro{Modelo: "Civic", Motor: motor}
	carro.ExibirDetalhes()
}

File: Lista/quest06/composicao.py
class Motor:
    def __init__(self, marca, potencia):
        self.marca = marca
        self.potencia = potencia

class Carro:
    def __init__(self, modelo, motor):
        self.modelo = modelo
        self.motor = motor

    def exibir_detalhes(self):
        print(f"Modelo: {self.modelo}, Motor: {self.motor.marca}, Potência: {self.motor.potencia} cv")

if __name__ == "__main__":
    motor = Motor("Honda", 150)
    carro = Carro("Civic", motor)
    carro.exibir_detalhes()
